drivers/sensors/apds9922.c

```diff
@@ -0,0 +1,2598 @@
+{
+  int ret;
+
+  ret = apds9922_i2c_write(priv, regaddr, &regval, 1);
+
+  return ret;
+}
+
+/****************************************************************************
+ * Name: apds9922_open
+ *
+ * Description:
+ *   Standard character driver close method.
+ *
+ * Input Parameters:
+ *   filep - file structure pointer
+ *
+ * Returned Value:
+ *   Success or failure
+ *
+ ****************************************************************************/
+
+static int apds9922_open(FAR struct file *filep)
+{
+  FAR struct inode *inode = filep->f_inode;
+  FAR struct apds9922_dev_s *priv = inode->i_private;
+  int ret;
+
+  ret = nxmutex_lock(&priv->devlock);
+  if (ret < 0)
+    {
+      return ret;
+    }
+
+  if (priv->crefs == 0)
+    {
+      priv->config->irq_attach(priv->config, apds9922_int_handler, priv);
+      priv->config->irq_enable(priv->config, true);
+    }
+
+  priv->crefs++;
+
+  nxmutex_unlock(&priv->devlock);
+
+  return OK;
+}
+
+/****************************************************************************
+ * Name: apds9922_close
+ *
+ * Description:
+ *   Standard character driver close method.
+ *
+ * Input Parameters:
+ *   filep - file structure pointer
+ *
+ * Returned Value:
+ *   Success or failure
+ *
+ ****************************************************************************/
+
+static int apds9922_close(FAR struct file *filep)
+{
+  FAR struct inode *inode = filep->f_inode;
+  FAR struct apds9922_dev_s *priv = inode->i_private;
+  int ret;
+
+  ret = nxmutex_lock(&priv->devlock);
+  if (ret < 0)
+    {
+      return ret;
+    }
+
+  DEBUGASSERT(priv->crefs > 0);
+
+  if (priv->crefs == 0)
+    {
+     apds9922_reset(priv);
+     apds9922_als_int_mode(priv, ALS_INT_MODE_DISABLED);
+     apds9922_ps_int_mode(priv, PS_INT_MODE_DISABLED);
+     priv->config->irq_detach(priv->config);
+    }
+
+  nxmutex_unlock(&priv->devlock);
+
+  return OK;
+}
+
+/****************************************************************************
+ * Name: apds9922_als_read
+ *
+ * Description:
+ *   Standard character driver read method.
+ *
+ * Input Parameters:
+ *   filep - File structure pointer
+ *   buffer - Buffer to write
+ *   buflen - The write length of the buffer
+ *
+ * Returned Value:
+ *   Size of buffer read
+ *
+ ****************************************************************************/
+
+static ssize_t apds9922_als_read(FAR struct file *filep, FAR char *buffer,
+                                size_t buflen)
+{
+  FAR struct inode *inode;
+
+  FAR struct apds9922_dev_s *priv;
+  int *ptr;
+  int ret;
+
+  DEBUGASSERT(filep);
+
+  inode = filep->f_inode;
+  priv = inode->i_private;
+
+  DEBUGASSERT(inode && inode->i_private);
+
+  ret = nxmutex_lock(&priv->devlock);
+  if (ret < 0)
+    {
+      return ret;
+    }
+
+  if (buflen < 1)
+    {
+      snerr("ERROR: Buffer not large enough to read data\n");
+      return (ssize_t)-EINVAL;
+    }
+
+  ptr = (int *)buffer;
```

Comment:
```suggestion
  ptr = (FAR int *)buffer;
```
How are we sure that `buffer` points to `int` size aligned memory? Should we state this in function description?